lib/matplotlib/backends/qt_compat.py

```diff
@@ -109,9 +109,18 @@
         except:
             res = 'QVariant API v2 specification failed. Defaulting to v1.'
             verbose.report(cond + res, 'helpful')
+    if QT_API == QT_API_PYQT5:
+        try:
+            from PyQt5 import QtCore, QtGui, QtWidgets
+            _getSaveFileName = QtWidgets.QFileDialog.getSaveFileName
+        except ImportError:
+            # fell through, tried PyQt5, failed fall back to PyQt4
+            QT_API = rcParams['backend.qt4']
```

Comment:
You can also get here if your rcparam is tkagg and you do not have qt5
installed (which is how I found this).

The (new) logic above says that if the rcparam is not set to a qt flavor,
it looks to see if qt4 or pyside is imported, then assumes it should be
using qt5.

On Mon, Sep 28, 2015, 04:30 Jens Hedegaard Nielsen <notifications@github.com>
wrote:

> In lib/matplotlib/backends/qt_compat.py
> <https://github.com/matplotlib/matplotlib/pull/5153#discussion_r40527986>:
>
> > @@ -109,9 +109,18 @@
> >          except:
> >              res = 'QVariant API v2 specification failed. Defaulting to v1.'
> >              verbose.report(cond + res, 'helpful')
> > +    if QT_API == QT_API_PYQT5:
> > +        try:
> > +            from PyQt5 import QtCore, QtGui, QtWidgets
> > +            _getSaveFileName = QtWidgets.QFileDialog.getSaveFileName
> > +        except ImportError:
> > +            # fell through, tried PyQt5, failed fall back to PyQt4
> > +            QT_API = rcParams['backend.qt4']
>
> Perhaps I misunderstand, but as far as I can see that will only happen if
>
>    - You have specified rcParams['backend.qt4'] = PyQt5 which is clearly
>    wrong.
>    - Tried using the Qt5Agg backend without installing PyQt5
>    - if if "PyQt4" in sys.modules or "PySide" in sys.modules: fails if
>    which case this will fail any way.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5153/files#r40527986>.
>